visitors/visitor.go

```diff
@@ -47,6 +47,11 @@ func (c ContextPath) GetLast() ContextPathElement {
 	return c[len(c)-1]
 }
 
+// Returns the `n`-th last element in the path.
+func (c ContextPath) GetNthLast(n int) ContextPathElement {
```

Comment:
Can you add a check here and return `nil` if there's an array underflow?

Reply:
In 94578a6.